shared/constants/route-tree.js

```diff
@@ -44,6 +47,7 @@ export type NavigateActions =
   | ResetRoute
 
 export const State = I.Record({
+  routeChanged: false,
```

Comment:
We want routeChanged to be separate from routeState, so it's not set back to false if routeState is reset. I want to try to avoid passing around a bunch of bare booleans for this, though.